analytics/build/build_test.go

```diff
@@ -359,78 +359,140 @@ func getActivityConfig(componentName string, allowReportAnalytics, allowTransmit
 }
 
 type mockAnalytics struct {
-	lastLoggedAuctionObject    *analytics.AuctionObject
-	lastLoggedAmpObject        *analytics.AmpObject
-	lastLoggedCookieSyncObject *analytics.CookieSyncObject
-	lastLoggedSetUIDObject     *analytics.SetUIDObject
-	lastLoggedVideoObject      *analytics.VideoObject
-	lastLoggedEventObject      *analytics.NotificationEvent
+	lastLoggedAuctionBidRequest *openrtb2.BidRequest
+	lastLoggedAmpBidRequest     *openrtb2.BidRequest
+	lastLoggedVideoBidRequest   *openrtb2.BidRequest
 }
 
 func (m *mockAnalytics) LogAuctionObject(ao *analytics.AuctionObject) {
-	m.lastLoggedAuctionObject = ao
+	m.lastLoggedAuctionBidRequest = ao.RequestWrapper.BidRequest
 }
 
 func (m *mockAnalytics) LogAmpObject(ao *analytics.AmpObject) {
-	m.lastLoggedAmpObject = ao
+	m.lastLoggedAmpBidRequest = ao.RequestWrapper.BidRequest
 }
 
-func (m *mockAnalytics) LogCookieSyncObject(ao *analytics.CookieSyncObject) {
-	m.lastLoggedCookieSyncObject = ao
+func (m *mockAnalytics) LogVideoObject(vo *analytics.VideoObject) {
+	m.lastLoggedVideoBidRequest = vo.RequestWrapper.BidRequest
 }
 
-func (m *mockAnalytics) LogSetUIDObject(ao *analytics.SetUIDObject) {
-	m.lastLoggedSetUIDObject = ao
+func (m *mockAnalytics) LogCookieSyncObject(ao *analytics.CookieSyncObject) {}
+
+func (m *mockAnalytics) LogSetUIDObject(ao *analytics.SetUIDObject) {}
+
+func (m *mockAnalytics) LogNotificationEventObject(ao *analytics.NotificationEvent) {}
+
+type secondMockAnalytics struct {
+	lastLoggedAuctionBidRequest *openrtb2.BidRequest
+	lastLoggedAmpBidRequest     *openrtb2.BidRequest
+	lastLoggedVideoBidRequest   *openrtb2.BidRequest
 }
 
-func (m *mockAnalytics) LogVideoObject(ao *analytics.VideoObject) {
-	m.lastLoggedVideoObject = ao
+func (m *secondMockAnalytics) LogAuctionObject(ao *analytics.AuctionObject) {
+	m.lastLoggedAuctionBidRequest = ao.RequestWrapper.BidRequest
 }
 
-func (m *mockAnalytics) LogNotificationEventObject(ao *analytics.NotificationEvent) {
-	m.lastLoggedEventObject = ao
+func (m *secondMockAnalytics) LogAmpObject(ao *analytics.AmpObject) {
+	m.lastLoggedAmpBidRequest = ao.RequestWrapper.BidRequest
 }
 
-func TestLogAnalyticsObjectExtRequestPrebidAnalytics(t *testing.T) {
+func (m *secondMockAnalytics) LogVideoObject(vo *analytics.VideoObject) {
```

Comment:
Why do you have two independent struct types rather than one type with two independent instances?